### ISSUE TITLE
Mexican balancing authority labeled as Canada in BA_Codes_930.xlsx

### ISSUE THREAD

**Issue by dt-woods** [USEPA/ElectricityLCI#234]
Inexplicably, Comision Federal de Electricidad (CFE) is listed under the Canadian balancing authorities (see 'Canada' worksheet in [BA_Codes_930.xlsx](https://github.com/USEPA/ElectricityLCI/blob/master/electricitylci/data/BA_Codes_930.xlsx)).

**Comment by m-jamieson** [comment]
I agree that can be changed - can either add a new sheet for Mexico or change the "Canada" sheet to "International" or something. Are we updating this for the 2020+ results?

**Comment by dt-woods** [comment]
My personal preference is move away from Microsoft Excel-based data files, which present a certain accessibility limitation and are not particularly well suited for git-based repositories (as they are stored as binaries). 

For a 2020+ update, what do you suggest? I could see a potential API call or data file(s) with balancing authority names.

**Comment by m-jamieson** [comment]
I don't think we're quite ready for API calls in the elci right? I know it's coming, but right now, I would want to avoid users having to get their own API key to use elci. This might be the last version for that though.

Having said all that, separate data files is okay with me.

**Comment by dt-woods** [comment]
> I don't think we're quite ready for API calls in the elci right? I know it's coming, but right now, I would want to avoid users having to get their own API key to use elci. This might be the last version for that though.
> 
> Having said all that, separate data files is okay with me.

Where did you get this EIA Operations Report in BA_Codes_930.xlsx?

Best I could find is the "[EIA-930 data reference tables](https://www.eia.gov/electricity/930-content/EIA930_Reference_Tables.xlsx)" in the About tab here: https://www.eia.gov/electricity/gridmonitor/about

I'm not finding a great resource for the annual BA codes in operation.

@bl-young, can StEWI return a list of balancing authority names, codes (abbreviations), and country of origin for a given year?

**Comment by bl-young** [comment]
> @bl-young, can StEWI return a list of balancing authority names, codes (abbreviations), and country of origin for a given year?

we don't currently access any EIA data in StEWI, only egrid (which I don't think has this information)

**Comment by dt-woods** [comment]
For completeness, see also, this commit with new BA code method: https://github.com/KeyLogicLCA/ElectricityLCI/commit/2ebda299782e9853487fab68dc8a5ca9761bbd17.

The new method could use an offline version of the Excel workbook (e.g., download to the new netl directory in electricitylci user data folder) and potentially a fallback to the old Excel workbook, which should eventually be phased out. The `create_ba_region_map` method needs linked to the new BA code method.

**Comment by dt-woods** [comment]
Note that `create_ba_region_map` is only in `eia860_balancing_authority` method found in eia860_facilities.py and only called when the method receives a 'regional_aggregation' parameter. `eia860_balancing_authority` is called in combinator.py, eia_io_trading.py, generation.py, and hydro_upstream.py; albeit none of these send regional aggregation parameter; therefore `create_ba_region_map` is likely unused.
